# SIT205 Thinking Systems and Cognition Science - Assignment 2

## Group: Philip Castiglione (217157862) and Warwick Smith (215239649)

## Topic 1: Text Analysis

## Report

This report satisfies the requirements for SIT205 Assignment 2, using the text analytics topic.

The code associated with this report is contained in a Jupyter Notebook named `SIT205 Assignment 2 - Project Code`. This report, the code, and associated data files can be found on GitHub at this link: 

https://github.com/PhilipCastiglione/SIT205_Watson

## Data

For this report we chose to analyse the recent events of Australian politics using text produced by members of the general public on Twitter.

Using the Twitter API, we collected 6,078 tweets containing the hashtag #libspill, made by 2,534 unique twitter user accounts. These tweets are dated between Saturday, 8th September and Monday, 17th September in 2018.

Tweets were cleaned of excess whitespace, tokenized and stripped of urls/links, @ mentions, and hashtags. Tweets were further filtered, excluding empty tweets, retweets and short tweets (less than 41 characters). The reduced set of 941 documents was our final corpus for textual analysis.

## Analysis

We analysed the tweets using the IBM Watson natural language understanding APIs.

In [26]:
import pickle
with open("cached_report_analysis_libspill.pkl", 'rb') as f:
    report_analyses = pickle.load(f)

#### Sentiment

We found the percentage of positive, netural and negative documents in the corpus:

In [27]:
print("Positive percentage:\t{:.2f}%".format(report_analyses['sentiment']['positive_percentage']))
print("Neutral percentage:\t{:.2f}%".format(report_analyses['sentiment']['neutral_percentage']))
print("Negative percentage:\t{:.2f}%".format(report_analyses['sentiment']['negative_percentage']))
print("Ratio of negative to positive:\t{:.2f}".format(
    report_analyses['sentiment']['negative_percentage'] / report_analyses['sentiment']['positive_percentage'])
)

Positive percentage:	17.53%
Neutral percentage:	22.95%
Negative percentage:	59.51%
Ratio of negative to positive:	3.39


The ratio of negative to positive sentiments of 3.39 in the corpus expresses a generally negative sentiment in tweets with the hashtag #libspill contained in our corpus.

We found the average and standard deviation of the positive and negative scores in the corpus:

In [28]:
print("Positive scores average:\t{:.3f}%\t (Std. Dev. = {:.3f})".format(
    report_analyses['sentiment']['average_pos_score'], report_analyses['sentiment']['std_dev_pos_score'])
)
print("Negative scores average:\t{:.3f}%\t (Std. Dev. = {:.3f})".format(
    report_analyses['sentiment']['average_neg_score'], report_analyses['sentiment']['std_dev_neg_score'])
)

Positive scores average:	0.571%	 (Std. Dev. = 0.288)
Negative scores average:	-0.618%	 (Std. Dev. = 0.204)


#### Emotion

In [29]:
# TODO
report_analyses['emotion']

{'sadness': {'average_score': 0.2915363602550483,
  'score_std_dev': 0.17761411949011788},
 'joy': {'average_score': 0.19376584697130708,
  'score_std_dev': 0.20529383029355744},
 'fear': {'average_score': 0.13972317428267853,
  'score_std_dev': 0.11275930185590587},
 'disgust': {'average_score': 0.2401355781083949,
  'score_std_dev': 0.1887589515629783},
 'anger': {'average_score': 0.22456241976620558,
  'score_std_dev': 0.1587676644763662}}

Notes:
not sure we really want the extra flavour around the ratio and stuff, just thought it was interesting. It's clearly outside of the assignment scope though, so possibly should remove it.